### PR TITLE
Add testcases for time window filter cache entry

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**
## Why this PR?

Added testcases to check whether the cache of time window filter cached the correct closest active recurring time window start.

https://github.com/microsoft/FeatureManagement-Dotnet/pull/266#discussion_r1593335729

### CODE REVIEW

**Review by ivywei0125 — APPROVED** [review]
Looks good to me

**Comment by jimmyca15** [comment]
The test doesn't appear to be validating what it intends to. Ensuring something is in the cache does not check whether the filter calculates the window again or not.

**Comment by zhiyuanliang-ms** [comment]
@jimmyca15 In the latest commit, I added an internal property for the timewindow filter to track how many times recurring time window is re-calculated. What do you think?

**Comment by zhiyuanliang-ms** [comment]
This PR goes after PR #452

**Comment by jimmyca15** [comment]
> @jimmyca15 In the latest commit, I added an internal property for the timewindow filter to track how many times recurring time window is re-calculated. What do you think?

@zhiyuanliang-ms too invasive. Can we pass a custom implementation of IMemoryCache and check whenever memorycache.Set is called?

**Comment by zhiyuanliang-ms** [comment]
> Can we pass a custom implementation of IMemoryCache and check whenever memorycache.Set is called?

@jimmyca15 Updated in [5da2d7d](https://github.com/microsoft/FeatureManagement-Dotnet/pull/445/commits/5da2d7d7ff89348ab7debeb52f9b158e9cd6ee50)

**Comment by zhiyuanliang-ms** [comment]
@rossgrambo Can you approve this PR?

File: tests/Tests.FeatureManagement/RecurrenceEvaluation.cs [hunk 1]
```diff
@@ -1644,43 +1644,80 @@ public async Task RecurrenceEvaluationThroughCacheTest()
...
+                DateTimeOffset? closestStart;
+                Assert.False(cache.TryGetValue(settings, out closestStart));
+                Assert.Equal(0, cache.CountOfEntryCreation);
 
-            for (int i = 0; i < 12; i++)
-            {
-                mockedTimeProvider.UtcNow = mockedTimeProvider.UtcNow.AddHours(1);
-                Assert.True(await mockedTimeWindowFilter.EvaluateAsync(context));
-            }
+                mockedTimeProvider.UtcNow = DateTimeOffset.Parse("2024-2-2T23:00:00+08:00");
```
[the remark this answers is not in the document]
Comment (zhiyuanliang-ms) [reply tests/Tests.FeatureManagement/RecurrenceEvaluation.cs:1658]: I find it easier to see the date directly, I used to open a calendar and check whether a date hits a recurrence. With`"StartTimeStamp.PlusDays(1)"`, I need to do some extra calculation in my mind.

**Pull request merged**